.cloudbuild/prombench.yaml

```diff
@@ -0,0 +1,490 @@
+    name: bash
+    env:
+    - '_COMMENTER=$_COMMENTER'
+    - '_COMMENTER_ROLE=$_COMMENTER_ROLE'
+    - '_GIT_REPO=$_GIT_REPO'
+    - '_PR_COMMENT_ACTION=$_PR_COMMENT_ACTION'
+    - '_PR_COMMENT_TEXT=$_PR_COMMENT_TEXT'
+    - '_PR_NUMBER=$_PR_NUMBER'
+    secretEnv: ['GH_PAT']
+    # allowExitCodes is used to not mark the step as failed in case of early exit (when prombench conditions are not met).
+    # This allows to exit gracefully from the current step.
+    allowExitCodes: [1]
+    script: |
+      #!/usr/bin/env bash
+
+      # Checking the action on the PR (benchmark should trigger only when a new comment is created)
+      # Available actions are: created, deleted and edited -- https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads
+      if ! [ "${_PR_COMMENT_ACTION}" == "created" ]
+      then
+        echo "The PR comment must be created (not edited or deleted) to trigger the benchmark."
+        echo "true" > /workspace/EXIT_EARLY
+        exit 1
+      fi
+
+      # Ignore comments from gmp-bot (as it will lead to recursive webhook triggers when gmp-bot comments again on the PR)
+      if ([ "${_COMMENTER}" == "gmp-bot" ])
+      then
+        echo "The comment is made by the gmp-bot. Hence, it will be ignored."
+        echo "true" > /workspace/EXIT_EARLY
+        exit 1
+      fi
+
+      # Checking the PR comment's text
+      if ! [[ "${_PR_COMMENT_TEXT}" == \/prombench* ]]
+      then
+        echo "The PR's comment (${_PR_COMMENT_TEXT}) does not start with '/prombench'"
+        echo "true" > /workspace/EXIT_EARLY
+        exit 1
+      fi
+
+      # User Verification (https://docs.github.com/en/graphql/reference/enums#commentauthorassociation)
+      if ! ([ "${_COMMENTER_ROLE}" == "OWNER" ] || [ "${_COMMENTER_ROLE}" == "MEMBER" ] || [ "${_COMMENTER_ROLE}" == "COLLABORATOR" ])
+      then
+
+        GITHUB_ORG=$(echo "${_GIT_REPO}" | cut -d'/' -f1)
+        GITHUB_REPO=$(echo "${_GIT_REPO}" | cut -d'/' -f2)
+        export GITHUB_ORG="$${GITHUB_ORG}" && export GITHUB_REPO="$${GITHUB_REPO}"
+
+        COMMENT_MSG="${_COMMENTER} does not have admin or write privileges. Therefore cannot execute the benchmark command."
+        GITHUB_URL="https://api.github.com/repos/$${GITHUB_ORG}/$${GITHUB_REPO}/issues/${_PR_NUMBER}/comments"
+        echo "$${COMMENT_MSG}"
+        
+        # Commenting on the GitHub PR
+        curl --silent -L -X POST \
+          -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $${GH_PAT}" -H "X-GitHub-Api-Version: 2022-11-28" \
+          "$${GITHUB_URL}" \
+          -d '{"body": "'"$${COMMENT_MSG}"'"}'
+        
+        echo "true" > /workspace/EXIT_EARLY
+        exit 1
+      fi
+
+  - id: "Setting environment variables"
+    name: gcr.io/cloud-builders/gcloud
+    env:
+    - '_PR_COMMENT_TEXT=$_PR_COMMENT_TEXT'
+    allowExitCodes: [1]
+    entrypoint: "bash"
+    args:
+      - "-c"
+      - |
+        if [ -f "/workspace/EXIT_EARLY" ]
```

Comment:
Do we need this check for every global variable?

 What if we have a step to create files with default values. That way we know the file exists for sure.